### ISSUE TITLE
add pager.html.twig override

### ISSUE THREAD

**Issue by alisonjo315** [CU-CommunityApps/cwd_base#41]
Let's add a pager.html.twig template override, to remove the redundant `role="navigation"` attribute from pagers.

It's a [known Drupal issue](https://www.drupal.org/project/drupal/issues/2655794), but they can't fix it until Drupal 10, because they aren't [dropping support for IE11 until Drupal 10](https://www.drupal.org/node/3199540).

But we don't have to support IE11!!!!!!

Silly GitHub doesn't allow `twig` file attachments, so I'll post the override we created for Bright Beams in a comment.
* "Before" i.e. what the pager is like without the proposed change (pager at bottom `<nav>` has a redundant `role` attribute): https://dev-cddemo.pantheonsite.io/events
* "After" i.e. what the pager is like with the proposed change (pager at bottom, redundant `role` is gone): https://cbb.cornell.edu/about/events/archive

**:question: QUESTION:** Does there need to be a corresponding update to cwd_framework?

----
NOTE: If CD Demo PR `#117` is approved _([Drupal core patch: aria-current attribute on pagination links](https://github.com/CU-CommunityApps/CD-demo/pull/117))_, we'll want to make sure to grab the "patched" version of the `stable` theme pager template as our basis for this template override.

**Comment by alisonjo315** [comment]
** :warning: REMINDER:** If [CD Demo PR `#117`](https://github.com/CU-CommunityApps/CD-demo/pull/117) is approved, we should use the [patched version of @stable/navigation/pager.html.twig](https://github.com/CU-CommunityApps/CD-demo/blob/2657e08c68ed99dcbe915a6b5e4b8bd31d268fc5/web/core/themes/stable/templates/navigation/pager.html.twig) as the basis for this template override!
----
`cwd_cbb/templates/overrides/pager.html.twig`
```twig
{#
/**
 * @file
 * Theme override to display a pager.
 * ~ Remove redundant role="navigation" from nav.pager (fix "redundant aria
 *   attribute" WA issue).
 * ~ Other than that, it's a direct copy of @stable/navigation/pager.html.twig.
 *
 * @see template_preprocess_pager()
 */
#}
{% if items %}
  <nav class="pager" aria-labelledby="{{ heading_id }}">
    <h4 id="{{ heading_id }}" class="visually-hidden">{{ 'Pagination'|t }}</h4>
    <ul class="pager__items js-pager__items">
      {# Print first item if we are not on the first page. #}
      {% if items.first %}
        <li class="pager__item pager__item--first">
          <a href="{{ items.first.href }}" title="{{ 'Go to first page'|t }}"{{ items.first.attributes|without('href', 'title') }}>
            <span class="visually-hidden">{{ 'First page'|t }}</span>
            <span aria-hidden="true">{{ items.first.text|default('« First'|t) }}</span>
          </a>
        </li>
      {% endif %}
      {# Print previous item if we are not on the first page. #}
      {% if items.previous %}
        <li class="pager__item pager__item--previous">
          <a href="{{ items.previous.href }}" title="{{ 'Go to previous page'|t }}" rel="prev"{{ items.previous.attributes|without('href', 'title', 'rel') }}>
            <span class="visually-hidden">{{ 'Previous page'|t }}</span>
            <span aria-hidden="true">{{ items.previous.text|default('‹ Previous'|t) }}</span>
          </a>
        </li>
      {% endif %}
      {# Add an ellipsis if there are further previous pages. #}
      {% if ellipses.previous %}
        <li class="pager__item pager__item--ellipsis" role="presentation">&hellip;</li>
      {% endif %}
      {# Now generate the actual pager piece. #}
      {% for key, item in items.pages %}
        <li class="pager__item{{ current == key ? ' is-active' : '' }}">
          {% if current == key %}
            {% set title = 'Current page'|t %}
          {% else %}
            {% set title = 'Go to page @key'|t({'@key': key}) %}
          {% endif %}
          <a href="{{ item.href }}" title="{{ title }}"{{ item.attributes|without('href', 'title') }}>
            <span class="visually-hidden">
              {{ current == key ? 'Current page'|t : 'Page'|t }}
            </span>
            {{- key -}}
          </a>
        </li>
      {% endfor %}
      {# Add an ellipsis if there are further next pages. #}
      {% if ellipses.next %}
        <li class="pager__item pager__item--ellipsis" role="presentation">&hellip;</li>
      {% endif %}
      {# Print next item if we are not on the last page. #}
      {% if items.next %}
        <li class="pager__item pager__item--next">
          <a href="{{ items.next.href }}" title="{{ 'Go to next page'|t }}" rel="next"{{ items.next.attributes|without('href', 'title', 'rel') }}>
            <span class="visually-hidden">{{ 'Next page'|t }}</span>
            <span aria-hidden="true">{{ items.next.text|default('Next ›'|t) }}</span>
          </a>
        </li>
      {% endif %}
      {# Print last item if we are not on the last page. #}
      {% if items.last %}
        <li class="pager__item pager__item--last">
          <a href="{{ items.last.href }}" title="{{ 'Go to last page'|t }}"{{ items.last.attributes|without('href', 'title') }}>
            <span class="visually-hidden">{{ 'Last page'|t }}</span>
            <span aria-hidden="true">{{ items.last.text|default('Last »'|t) }}</span>
          </a>
        </li>
      {% endif %}
    </ul>
  </nav>
{% endif %}
```

Diff comparison, based on [@stable/navigation/pager.html.twig](https://github.com/CU-CommunityApps/CD-demo/blob/main/web/core/themes/stable/templates/navigation/pager.html.twig):
```diff
  */
 #}
 {% if items %}
-  <nav class="pager" role="navigation" aria-labelledby="{{ heading_id }}">
+  <nav class="pager" aria-labelledby="{{ heading_id }}">
     <h4 id="{{ heading_id }}" class="visually-hidden">{{ 'Pagination'|t }}</h4>
     <ul class="pager__items js-pager__items">
       {# Print first item if we are not on the first page. #}
```

**Comment by ama39** [comment]
Sure, go for it. It should be safe to remove the attribute. It doesn't interface with our CSS. Although... if we're going to be in the code anyway, I might recommend removing the `aria-labelledby` and the `h4` it references, and replacing it with a simple `aria-label`. Specific heading levels like this `h4` not ideal, since we don't know what the rest of the heading structure on the page might be. You could avoid this by changing it to:

`<nav class="pager" aria-label="{{ 'Pagination'|t }}">`

Of course, if there are multiple pagers on the page, there will be duplicate labels. But that was already a shortcoming with Drupal's default code.

**Comment by alisonjo315** [comment]
Discussed during May 8, 2023 full-CD-team meeting, decided to abandon this issue and its PR, because it's not a strict failure, and we're almost to Drupal 10 anyway, the fix will be coming to us via Drupal core.

I will create a follow-up issue to investigate the use `<h4>`  situation Tone pointed out.

**Comment by alisonjo315** [comment]
GOOD NEWS!  The heading level issue is already in-progress in Drupal, so I think we can punt on that, too:
https://www.drupal.org/project/drupal/issues/3333401
